NitroxServer/Serialization/ServerConfig.cs

```diff
@@ -41,6 +43,50 @@ public int MaxConnections
             }
         }
 
+        private GameModeOption? _gameMode = null;
+        public GameModeOption GameMode
+        {
+            get
+            {
+                if (_gameMode == null && ConfigurationManager.AppSettings[GAMEMODE_SETTING] != null)
+                {
+                    _gameMode = ParseGameMode(ConfigurationManager.AppSettings[GAMEMODE_SETTING]);
+                }
+                return _gameMode ?? DEFAULT_GAMEMODE;
+            }
+        }
+
+        GameModeOption ParseGameMode(string stringGameMode)
+        {
+            GameModeOption gameMode = GameModeOption.Survival;
+            stringGameMode = stringGameMode.ToLower(); // Lets be frank people have habits
+            switch (stringGameMode)
```

Comment:
Might be cleaner as a dictionary above:

```
Dictionary<string, GameModeOption> gameModeByConfigKey = new Dictionary<string, GameModeOption> {
  {"survival", GameModeOption.Survival},
  {"creative", GameModeOption.Creative},
  {"hardcore", GameModeOption.Hardcore},
  {"permadeath, GameModeOption.Survival | GameModeOption.Permadeath}
};
```

Then you can remove this method and simply use the map in the `GameMode` getter

Reply:
used TryGetValue after creating Dictionary